src/index.js

```diff
@@ -159,19 +165,19 @@ const ProgressButton = createReactClass({
     this.setState({currentState: STATE.DISABLED})
   },
 
-  success (callback, dontRemove) {
+  success (callback, stayInSuccessState = false) {
     this.setState({currentState: STATE.SUCCESS})
     this._timeout = setTimeout(() => {
-      if (!dontRemove) { this.setState({currentState: STATE.NOTHING}) }
+      if (!stayInSuccessState || !this.props.stayInSuccessState) { this.setState({currentState: STATE.NOTHING}) }
```

Comment:
I think you don't need the default argument and then the check becomes:
```js
if (typeof stayInSuccessState !== 'undefined' ? !stayInSuccessState : !this.props.stayInSuccessState) {
  ...
}
```
otherwise the argument won't ever be used if the prop is `true`